garak/generators/openai.py

```diff
@@ -282,4 +314,27 @@ def __init__(self, name="", config_root=_config):
         super().__init__(self.name, config_root=config_root)
 
 
+class ReasoningGenerator(OpenAIGenerator):
+    """Generator wrapper for OpenAI reasoning models, e.g. `o1` family."""
+
+    supports_multiple_generations = False
+
+    DEFAULT_PARAMS = Generator.DEFAULT_PARAMS | {
+        "top_p": 1.0,
+        "frequency_penalty": 0.0,
+        "presence_penalty": 0.0,
+        "seed": None,
+        "stop": ["#", ";"],
+        "suppressed_params": set(["n", "temperature", "max_tokens", "stop"]),
+        "retry_json": True,
+        "custom_params": {
+            "max_completion_tokens": 1500,
+        },
```

Comment:
Based on converting to checking signature of `create` for accepted params.
```suggestion
        "max_completion_tokens": 1500,
```